src/AsimovDeploy.WinAgentUpdater/AsimovConfigUpdate.cs

```diff
@@ -6,11 +6,17 @@ public class AsimovConfigUpdate
     {
         public string FilePath { get; set; }
         public int Version { get; set; }
+        public string Bucket { get; set; }
+        public string FileNameGS { get; set; }
+        public string FileName { get; set; }        
     }
 
     public class AsimovVersion
     {
-        public Version Version { get; set; }
         public string FilePath { get; set; }
+        public Version Version { get; set; }
+        public string Bucket { get; set; }
```

Comment:
I think you can get rid of this duplication by extracting how to locate the file to an interface. E.g.:

`public class AsimovVersion
    {
        public Version Version { get; set; }
        private IAsimovFileSource FileSource { get; set; }
    }


    public interface IAsimovFileSource
    {
        string GetFilePath();
    }

    public class FileSystemFileSource : IAsimovFileSource
    {
        private readonly string _path;

        public FileSystemFileSource(string path)
        {
            _path = path;
        }

        public string GetFilePath()
        {
            return _path;
        }
    }

    public class GcpAsimovFileSource : IAsimovFileSource
    {
        private readonly Object Object;

        public GcpAsimovFileSource(Object o)
        {
            Object = o;
        }

        public string GetFilePath()
        {
            var client = new StorageClient();
            var tempFile = Path.GetTempFileName();

            using (var fileStrem = File.OpenWrite(tempFile))
            {
                client.DownloadObject(Object, fileStrem);
            }
            return tempFile;


        }

    }`